### ISSUE TITLE
Add RunAsync() method

### ISSUE THREAD

**Issue by Ullaakut** [Ullaakut/nmap#6]
Allowing the users to run `nmap` asnchronously makes it possible to parse the live `nmap` output, for example for building a UI on top, or computing a progress bar.

I'm not sure if many people have this use-case though, so I won't prioritize this for now, but I find it really cool.

- [ ] Add a RunAsync() method with the following signature `func (s *Scanner) RunAsync() (stdout chan <-[]byte, stderr chan <-[]byte, error)`
	- [ ] Make this method return one channel for stdout, one for stderr, and an error if the `nmap` process could not be started.
	- [ ] Have goroutine to kill nmap if context is done or times out.
	- [ ] Store output in private attribute to expose a `nmap.Result` structure once the scan is done

**Comment by TheSecEng** [comment]
Have you had time to look at this more ? If not, I’ll start looking into how to add this properly. It’s be nice to have.

**Comment by Ullaakut** [comment]
Hey @TheSecEng I didn't have time as of now, I've been working on other projects, but if you're interested in picking this one up I'd love to give you any help you need :)

**Comment by TheSecEng** [comment]
@Ullaakut you got it!

**Comment by Ullaakut** [comment]
Hey @TheSecEng!
 
Sorry about the delay. This looks like a good implementation! Thank you so much!

It seems like a good first step to provide utility to people who might want to use it, but I'm curious how people will end up integrating it into their apps.

Either way, I'd love to see it in the form of a PR and to work on it together!

**Comment by TheSecEng** [comment]
> Hey @TheSecEng!
> 
> Sorry about the delay. This looks like a good implementation! Thank you so much!
> 
> It seems like a good first step to provide utility to people who might want to use it, but I'm curious how people will end up integrating it into their apps.
> 
> Either way, I'd love to see it in the form of a PR and to work on it together!

I've actually got a use case right now that I will be using this for once its finished. So there is definitely reasons to get this running.
